rust/core-lib/src/editor.rs

```diff
@@ -687,6 +688,38 @@ impl Editor {
         }
     }
 
+    fn duplicate_line(&mut self, view: &View, config: &BufferItems) {
+        let mut builder = delta::Builder::new(self.text.len());
+        // get affected lines
+        let mut lines = BTreeSet::new();
```

Comment:
This part is almost the same as in `modify_indent`. Perhaps there is a way to reuse the code...

Reply:
that could make sense at some point, feels like this is okay for now though.